### PR TITLE
Issue 2065

### LINKED ISSUE #2065: Deemphasize Clear layer button
# Question

image
Shouldn't we make the `Clear layer` button less prominent? I think its a bit too easy clickable

### LINKED ISSUE #2101: Drawing invisible features
No default styling is applied for features of certain geometry when SLD defined in app config doesnt include necessary symbolizer.

**To Reproduce**
1. Run test-app
2. Add polygon/line geometry feature to 'Points' layer

**Expected behavior**
All types of geometry should be visualized properly, proper symbolizer should be added to SLD/OL style if new gemetry is added to layer

### PR DESCRIPTION
**Pull request by FilipLeitner**
## Description

- Add possiblity to select features by 'polygon' 
- Add possiblity to dowload/ remove all selected features

## Related issues or pull requests
fixes #2065 #2062
closes #2101
## Pull request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [x] Feature
- [ ] Dependency updates
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe)
- [ ] I am unsure (we'll look into it together)

## Do you introduce a breaking change?

- [ ] Yes
- [x] No
- [ ] I am unsure (no worries, we'll find out)

## Checklist

- [ ] I understand and agree that the changes in this PR will be licensed under the [MIT License]
- [ ] I have followed the [guidelines for contributing](https://github.com/hslayers/hslayers-ng/blob/master/CONTRIBUTING.md)
- [ ] The proposed change fits to the content of the [code of conduct](https://github.com/hslayers/hslayers-ng/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have added or updated tests and documentation, and the test suite passes (run `npm test` locally)
- [ ] I'm lost; why do I have to check so many boxes? Please help!

### CODE REVIEW

**Comment by FilipLeitner** [comment]
Currenty  I set maximum of rendered query-feature components to be rendered to 50 which is still pretty high in my opinion, we could maybe make it even a bit lower.
Was also thinking whether 'Select all features' should work as a toggle.
What do you think?

File: projects/hslayers/src/css/app.scss [hunk 1]
```diff
@@ -156,6 +156,10 @@ $mobile-layout-breakpoint: $mobile-layout-breakpoint-value +'px' !default; //thi
 		font-size: x-large !important;
 	}
 
+	.cursor-pointer {
```
Comment (jmacura) [inline projects/hslayers/src/css/app.scss:159]: Wow, when did this go out? I thought we have that class present in SCSS for the whole time 😮

**Comment by jmacura** [comment]
> Was also thinking whether 'Select all features' should work as a toggle.
> What do you think?

I am very much in favour of changing it to 'Select all features'/'Deselect all features'. Beside it being more intuitive behaviour, it should also circumvent multi-selection of the same feature, which happens now:
image
(by clicking the 'Select all features' repeatedly, it gets selected repeatedly)

File: projects/hslayers/src/assets/locales/en.json [hunk 2]
```diff
@@ -396,7 +396,14 @@
...
     "snapTo": "Snap to features of layer",
-    "unsavedDrawing": "Unsaved drawing"
+    "unsavedDrawing": "Unsaved drawing",
+    "boxSelectionActivated": "Box selection activated",
+    "useModifierToSelectWithBox": "Use {{platformModifierKey}} and drag to select features with box",
+    "noSuitableLayer": "No suitable layer available",
+    "selectAllFeatures": "Select all features",
+    "boxSelection": "Box selection",
+    "stylingMissing": "Proper styling missing",
+    "stylinMissingWarning": "{{type}} geometry can NOT be visualized. Create {{symbolizer}} symbolizer first. You can do so in {{panel}} panel."
```
[the remark this answers is not in the document]
Comment (FilipLeitner) [reply projects/hslayers/src/assets/locales/en.json:406]: Took me a while to spot the difference even tho its highlighted, thanks

File: projects/hslayers/src/assets/locales/en.json [hunk 3]
```diff
@@ -583,7 +590,11 @@
     "reallyDelete": "Really delete this feature?",
     "reallyDeleteAllFeaturesFrom": "Really delete all features from layer {0}?",
     "untitledFeature": "Untitled feature",
-    "clusterContaining": "Cluster containing {{count}} features"
+    "clusterContaining": "Cluster containing {{count}} features",
+    "deleteAllSelected": "Delete all selected features",
+    "featuresSelected": "Features selected: {{featureCount}}",
```
[the remark this answers is not in the document]
Comment (FilipLeitner) [reply projects/hslayers/src/assets/locales/en.json:595]: Got carried away somehow and havent realized this is not necessary, will refactor

File: projects/hslayers/src/components/query/feature.component.ts [hunk 4]
```diff
@@ -28,24 +35,49 @@ export class HsQueryFeatureComponent {
...
+    public cd: ChangeDetectorRef
   ) {}
 
-  olFeature() {
+  ngAfterViewInit(): void {
+    this.availableLayersSubscription =
+      this.HsFeatureCommonService.availableLayer$.subscribe((layers) => {
+        this.availableLayers = layers.filter((layer) => {
+          return (
+            (layer as any).ol_uid !=
```
Comment (raitisbe) [inline projects/hslayers/src/components/query/feature.component.ts:58]: Could not compare the layer instances directly?

**Comment by raitisbe** [comment]
The functionality looks realy great, thank you!

**Comment by raitisbe** [comment]
Shouldn't we expand this by default?
image

**Comment by FilipLeitner** [comment]
> Shouldn't we expand this by default?
> image

Wasnt quite sure about this. Having it closed by default saves some computation power by not rendering the query feature components every time. Its probably not that much of difference but I just wanted to keep it a bit quicker. But if you feel its not convenient I can change that :)

**Comment by raitisbe** [comment]
> > Shouldn't we expand this by default?
> > image
> 
> Wasnt quite sure about this. Having it closed by default saves some computation power by not rendering the query feature components every time. Its probably not that much of difference but I just wanted to keep it a bit quicker. But if you feel its not convenient I can change that :)

Yes, the CPU cycles are cheap. Users attention is not ;)

**Review by raitisbe — APPROVED** [review]

**Pull request merged**